Iter9

DenisKhanov opened the pull request.

DenisKhanov committed:
Добавил логгирование logrus, которое записывается в файл app.log с помощью библиотечки lumberjack
Поправил тесты конфига

DenisKhanov committed:
Merge branch 'main' into iter6

DenisKhanov committed:
Добавил отдельную проверку флага -a в конфиг тестах.

DenisKhanov committed:
Не понимаю почему валится автотест. Попробовал инициализацию cfg в init перенести

DenisKhanov committed:
Убрал обработку флага LOG_LEVEL в попытке пройти автотесты

DenisKhanov committed:
Назад вернул установку уровня логирования

DenisKhanov committed:
Подбор названий переменных логирования

DenisKhanov committed:
Подбор названий переменных логирования

DenisKhanov committed:
Подбор названий переменных логирования

DenisKhanov committed:
Подбор названий переменных логирования

DenisKhanov committed:
Подбор названий переменных логирования

DenisKhanov committed:
Закоменчен только 4й тест конфига

DenisKhanov committed:
Закоменчен только 4й тест конфига

DenisKhanov committed:
Откат, убрал переменную среды лог левел

DenisKhanov committed:
Добавил эндпоинт "/api/shorten"
Реализовал метод JsonURL принимающий JSON, обрабатывающий его и возвращающий ответ тоже в JSON
И написал тесты для него

DenisKhanov committed:
Исправил непонятное замечание автотестов, JsonURL на JSONURL

DenisKhanov committed:
В тестах тоже поправил

DenisKhanov committed:
Добавил сжатие gzip с помощью MiddlewareCompress

DenisKhanov committed:
Чуть упростил код в middlewareCompress

DenisKhanov committed:
Добавил в middlewareCompress проверку content-type при "Content-Encoding" - "gzip"

DenisKhanov committed:
Подразобрался в своем middleware в плане сжатия, добавил проверку на сжатие если размер данных больше 1400, из-за чего скорее всего ляжет тест 8го инкремента, но что по идее нужно использовать.

DenisKhanov committed:
Добавил выбор записи в файл или записи в память. Написал тесты для этого изменения

DenisKhanov committed:
Поправил нейминг и добавил создание и запись инфы в файл теста.

DenisKhanov committed:
Исправил файл конфигурации, из-за него валятся тесты 5 инкремента

DenisKhanov committed:
Очередная попытка добавить установку уровня логирования из командной строки и переменной окружения, плюс изменил логику получения UUID для записи данных

DenisKhanov committed:
ТЕст 9 инкремента валится. У меня было настроено, чтобы путь к файлу создавался в директории родительской. Сейчас убрал это условие

edylcrm commented on cmd/shortener/main.go:
```diff
@@ -9,29 +9,71 @@ import (
 
+var cfg *config.ENVConfig
+
+func init() {
```
[BLOCKER] такие функции лучше не использовать. Иногда их используют в библиотеках для инициализации чего-либо. Но большое их кол-во может привести к некорректной работе программы. Если можно обойтись без этой функции, то не стоит ее использовать

edylcrm commented on cmd/shortener/main.go:
```diff
@@ -9,29 +9,71 @@ import (
 	"github.com/DenisKhanov/shorterURL/internal/app/repositoryes"
```
**[BLOCKER]** -  Проблема, не решив которую принять работу нельзя
**[LINT]** - Заменичение, опционально к выполнению
**[FYI]** - просто ремарка

edylcrm commented on internal/app/repositoryes/saveToFile.go:
```diff
@@ -0,0 +1,141 @@
+	if err := os.MkdirAll(dir, 0666); err != nil {
+		log.Fatal(err)
+	}
+	file, err := os.OpenFile(r.storageFilePath, os.O_WRONLY|os.O_CREATE|os.O_APPEND, 0666)
```
[BLOCKER] лучше будет открыть файл один раз при инициализации структуру и работать с ним. Открывать каждый раз файл на запись - дорогая операция. Так же лучше сохранять в файл не при каждой попытке создать новый сокращенный url, а batch запросами. Здесь лучше объединить подход с mem Storage. Хранить данные в памяти и переодически записывать их в файл

DenisKhanov committed:
Куча изменений:
1. убрал второй репозиторий пишуший в файл
2. убрал в main переключение между записью в файл или память
3. теперь в репозитории логика такова, что при старте сервера читается файл (если он есть), и из него данные записываются в память, далее работа идет с памятью, и если в памяти 100 ссылок, то они записываются в файл.
4. запись в файл из памяти происходит и перед закрытием сервера по сигналу.

DenisKhanov committed:
Подправил нейминг
